cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/cube/DefaultCube.java

```diff
@@ -140,19 +140,30 @@ SELECT count('read.ops')                                     << measure name and
                                            query.toString());
     }
 
-    Aggregation agg = findAggregation(query);
-    if (agg == null) {
-      throw new IllegalArgumentException("There's no data aggregated for specified tags to satisfy the query: " +
-                                           query.toString());
+    // 1) find aggregation to query
+    Aggregation agg;
+    if (query.getAggregation() != null) {
+      agg = aggregations.get(query.getAggregation());
+      if (agg == null) {
+        throw new IllegalArgumentException("Specified aggregation does not exist in cube: " + query.getAggregation());
```

Comment:
should we list the available aggregation names here ? , so its useful for the user if there is a typo.